### PR TITLE
Order public communities by "hotness" by default

### PR DESCRIPTION
**Pull request by megrogan**

### CODE REVIEW

**Comment by megrogan** [comment]
@hpeebles quite a lot to discuss in this PR...

File: backend/canisters/group_index/impl/src/model/public_groups.rs [hunk 1]
```diff
@@ -151,13 +149,12 @@ impl PublicGroups {
...
 
-    pub fn calculate_hot_groups(&self, now: TimestampMillis) -> Vec<ChatId> {
-        let mut rng = StdRng::seed_from_u64(now);
+    pub fn calculate_hot_groups(&self, now: TimestampMillis, random: u32) -> Vec<ChatId> {
         let one_day_ago = now - DAY_IN_MS;
 
         self.iter()
             .filter(|g| !g.is_frozen() && g.has_been_active_since(one_day_ago) && !g.exclude_from_hotlist)
-            .map(|g| (g, rng.next_u32()))
+            .map(|g| (g, random))
```
Comment (hpeebles) [inline backend/canisters/group_index/impl/src/model/public_groups.rs:157]: This passes the same value to each group so it isn't really adding any randomness

Comment (megrogan) [reply backend/canisters/group_index/impl/src/model/public_groups.rs]: Doh!

File: backend/canisters/community/impl/src/activity_notifications.rs [hunk 2]
```diff
@@ -23,6 +22,96 @@ pub(crate) fn handle_activity_notification(state: &mut RuntimeState) {
...
+
+        let mut message_unique_users = HashSet::new();
+        let mut reaction_unique_users = HashSet::new();
+
+        for event in data
+            .channels
+            .iter()
+            .filter(|channel| channel.chat.is_public)
+            .filter_map(|channel| {
+                channel.chat.events.events_list(EventIndex::default(), None, now).map(|list| {
```
Comment (hpeebles) [inline backend/canisters/community/impl/src/activity_notifications.rs:76]: I think you can use `main_events_reader` here instead.

Comment (megrogan) [reply backend/canisters/community/impl/src/activity_notifications.rs:76]: Having borrowing issues trying to use a reader

Comment (megrogan) [reply backend/canisters/community/impl/src/activity_notifications.rs:76]: Let's discuss tomorrow

**Review by hpeebles — APPROVED** [review]

**Pull request merged**